src/env.cc

```diff
@@ -231,6 +253,23 @@ void Environment::Start(const std::vector<std::string>& args,
   HandleScope handle_scope(isolate());
   Context::Scope context_scope(context());
 
+  if (*TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED(
+      TRACING_CATEGORY_NODE1(environment))) {
+    auto traced_value = tracing::TracedValue::Create();
+    traced_value->BeginArray("args");
+    for (std::string arg : args)
+      traced_value->AppendString(arg);
+    traced_value->EndArray();
+    traced_value->BeginArray("exec_args");
+    for (std::string arg : exec_args)
+      traced_value->AppendString(arg);
+    traced_value->EndArray();
+    TRACE_EVENT_NESTABLE_ASYNC_BEGIN1(
+      TRACING_CATEGORY_NODE1(environment),
+      "Environment", this,
+      "args", std::move(traced_value));
+  }
```

Comment:
Heads up … there’s a very good chance of the `args`/`exec_args` handling being moved to another method as part of better embedding support

Reply:
Yeah, I was wondering about that. Hmm... perhaps we should make args have their own dedicated trace event category... or even make it part of the process `__metadata` so that it's always there if tracing is enabled. It's useful context either way.

Reply:
@jasnell I think that depends… We’ve had a similar conversation in the node-report PR at https://github.com/nodejs/node/pull/22712 – what arguments do we actually care about here? The ones passed on the original command line, to the Node.js CLI, or the one that this Node.js instance explicitly uses?

The distinction matters for embedder use cases; But maybe, as a more practical example: I’d like to add support for something similar to `execArgv` to Workers. I guess it makes sense to emit these per-Environment?

Reply:
I'm thinking that the one that this Node.js instance explicitly uses... whichever are relevant to each specific `Environment` instance.